keps/sig-node/3960-pod-lifecycle-sleep-action/README.md

```diff
@@ -323,6 +329,16 @@ Previously created pod with sleep hook set will execute it before terminating.
 
 Yes. Some unit tests will be designed to test the verification process of the "sleep" field under different scenarios, such as when the feature is enabled, disabled, or switched. These tests will be included in the alpha version.
 
+Added in alpha:
+Unit tests in `k8s.io/kubernetes/pkg/apis/core/validation_test` test the verification with feature-gate enabled.
+The verification with feature-gate disabled is tested manually as it's hard to trigger `PrepareForCreate` strategy in unit tests.
+E2e tests in `test/e2e/common/node/lifecycle_hook.go` test the behavior with feature-gate enabled.
```

Comment:
Copying from the KEP template:
```
Additionally, for features that are introducing a new API field, unit tests that
are exercising the `switch` of feature gate itself (what happens if I disable a
feature gate after having objects written with the new field) are also critical.
You can take a look at one potential example of such test in:
https://github.com/kubernetes/kubernetes/pull/97058/files#diff-7826f7adbc1996a05ab52e3f5f02429e94b68ce6bce0dc534d1be636154fded3R246-R282
```

I'm actually more interested in sth like that - this question is not about testing the feature itself (testing code with FG enabled), but rather testing the transition between enabled and disabled (and vice versa).

Reply:
... Yea, I think we need a test like this for this feature. But I clearly misunderstood this and missed this in alpha🤦.
I'll bring this test in beta, I added some cases in `Unit tests` for beta.